File: backend/TransportStatic/Models/Shape.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using NetTopologySuite.Geometries;

namespace TransportStatic.Models;

[Table("shapes")]
public class Shape
{
    [Column("shape_id")]
    public string Id { get; set; } = null!;

    [Column("shape_pt_lat")]
    public decimal Latitude { get; set; }

    [Column("shape_pt_lon")]
    public decimal Longitude { get; set; }

    [Column("geom")]
    public Point Geom { get; set; } = null!;

    [Column("shape_pt_sequence")]
    public int Sequence { get; set; }

    [Column("shape_dist_travelled")]
    public decimal? DistanceTravelled { get; set; }
    
    [Column("mode")]
    public string Mode { get; set; } = null!;
}

File: backend/TransportRealtime/models/static/shape.go
package models

import "github.com/twpayne/go-geom"

type Shape struct {
	ShapeId            string      `json:"shapeId"`
	ShapePtLat         float64     `json:"shapePtLat"`
	ShapePtLon         float64     `json:"shapePtLon"`
	ShapeGeom          *geom.Point `json:"shapeGeom"`
	ShapePtSequence    int         `json:"shapePtSequence"`
	ShapeDistTravelled *float64    `json:"shapeDistTravelled"`
	Mode               string      `json:"mode"`
}

type ShapeDetails struct {
	ShapePtLat         float64     `json:"shapePtLat"`
	ShapePtLon         float64     `json:"shapePtLon"`
	ShapeGeom          *geom.Point `json:"shapeGeom"`
	ShapePtSequence    int         `json:"shapePtSequence"`
	ShapeDistTravelled *float64    `json:"shapeDistTravelled"`
	Mode               string      `json:"mode"`
}
